components/roadmap-grid/RoadmapDetailedView.tsx

```diff
@@ -55,71 +56,72 @@ export function RoadmapDetailed({
   const [numHeaderTicks, setNumHeaderTicks] = useState(5);
   const [numGridCols, setNumGridCols] = useState(45);
 
-
-  const today = dayjs();
-
+  // for preventing dayjsDates from being recalculated if it doesn't need to be
+  const issuesGroupedId = issuesGroupedState.value.map((g) => g.groupName).join(',');
   /**
    * Collect all due dates from all issues, as DayJS dates.
    */
-  let dayjsDates: Dayjs[] = []
-  // useEffect(() => {
+  const dayjsDates: Dayjs[] = useMemo(() => {
+    const today = dayjs();
+    let innerDayjsDates: Dayjs[] = []
     try {
-      dayjsDates = issuesGroupedState.value
+      innerDayjsDates = issuesGroupedState.value
         .flatMap((group) => group.items.map((item) => dayjs(item.due_date).utc()))
         .filter((d) => d.isValid());
```

Comment:
we should update this to parse over issueData and all children instead of the grouped object... all we need are the dates